apps/evm/lib/evm/gas.ex

```diff
@@ -351,28 +351,44 @@ defmodule EVM.Gas do
     iex> exec_env = %EVM.ExecEnv{address: address, account_interface: account_interface}
     iex> EVM.Gas.operation_cost(:sstore, [0, 0], %EVM.MachineState{}, exec_env)
     5000
-    iex> EVM.Gas.operation_cost(:sstore, [0, 2], %EVM.MachineState{}, exec_env)
-    20000
   """
-  def operation_cost(:sstore, [key, new_value], _machine_state, exec_env) do
-    old_value = ExecEnv.get_storage(exec_env, key)
-
-    if old_value == 0 && new_value != 0 do
-      @g_sset
-    else
-      @g_sreset
+  def operation_cost(:sstore, params = [key, new_value], _machine_state, exec_env) do
+    case ExecEnv.get_storage(exec_env, key) do
```

Comment:
I wish we could find a way to simplify this branching logic but it looks like this is as good as we are going to get 😕